packages/tools/viewer/src/viewer.ts

```diff
@@ -641,13 +802,24 @@ export class Viewer implements IDisposable {
                     camera.radius = distance;
                     camera.interpolateTo(undefined, undefined, undefined, pickingInfo.pickedPoint);
                 } else {
-                    camera.restoreState();
+                    this.resetCamera(true);
                 }
             }, PointerEventTypes.POINTERDOUBLETAP);
 
+            scene.onNewCameraAddedObservable.add((camera) => {
+                if (this.camerasAsHotSpots) {
+                    this._addCameraHotSpot(camera, this._camerasAsHotSpotsAbortController?.signal);

```

Comment:
a floating promise inside the observer. We do that from time to time in other places, but we should probably avoid doing that as much as possible, because of difficult error handling/tracing. Or handle errors right here :-)

Reply:
I'll wrap this in the same `handleError` helper function.